client/ayon_applications/plugins/launcher_actions/debug_terminal.py

```diff
@@ -0,0 +1,178 @@
+import os
+from typing import Optional
+
+from qtpy import QtWidgets, QtGui, QtCore
+
+from ayon_applications import (
+    Application,
+    ApplicationManager,
+    APPLICATIONS_ADDON_ROOT
+)
+from ayon_applications.utils import get_app_environments_for_context
+from ayon_core.pipeline import LauncherAction
+from ayon_core.style import load_stylesheet
+from ayon_core.tools.utils.lib import get_qt_icon
+
+
+def get_application_qt_icon(application: Application) -> Optional[QtGui.QIcon]:
```

Comment:
Just reminder, there already is function to get icon path, and there should be only one. I would move it from addon method to `utils.py` so it can be used anywhere.

Reply:
Lovely - where is the 'already existing one' that I should be adopting?

Reply:
Did you mean `get_app_icon_path` that you mentioned [here](https://github.com/ynput/ayon-applications/pull/12#discussion_r1738852629)?

This seems to be [here](https://github.com/ynput/ayon-applications/blob/144163d05a2a714f11e807fa0d29d3d7feca88c8/client/ayon_applications/addon.py#L122-L138) on the Applications addon object. From the Launcher action, what is the best efficient way to access the addon instance? Do I need to initialize a full `AddonsManager` instance?

Reply:
Yes, that's why I would move the logic to `utils.py` and import/use it where is needed.

Reply:
Implemented with https://github.com/ynput/ayon-applications/pull/12/commits/d41e43ee1c715ed3da074c5afeb9440b3df8f1fc **but** the `get_application_qt_icon` logic is still more involved and as you said may warrant moving to that `utils`. Moving that to `utils` may mean that the Addon itself also imports `qtpy` - which is likely not what we want?

Also, I just want to emphasize that there's **confusingly** also a `get_app_icon_path` function in `ayon_core.style`, see: https://github.com/ynput/ayon-core/blob/0701238b531f052bb526429186c263ac31fa1c38/client/ayon_core/style/__init__.py#L208-L210

I guess we'll still need to refactor/cleanup some more. Any pointers on how you'd like me to proceed?

Reply:
I didn't mean `get_application_qt_icon`, I meant just the 

> Also, I just want to emphasize that there's confusingly also a get_app_icon_path function in ayon_core.style, see: https://github.com/ynput/ayon-core/blob/0701238b531f052bb526429186c263ac31fa1c38/client/ayon_core/style/__init__.py#L208-L210

Should be removed in future, it is there for backwards compatibility.